modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-volley/request/IRequestFactory.mustache

```diff
@@ -0,0 +1,64 @@
+package {{packageName}}.request
+
+import com.android.volley.Request
+import com.android.volley.Response
+import java.io.UnsupportedEncodingException
+import java.lang.reflect.Type
+import java.net.URLEncoder
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.*
+import java.time.format.DateTimeFormatter
+import java.time.OffsetDateTime
+import java.time.LocalDate
+
+
+interface IRequestFactory {
+
+    companion object {
+        /**
+         * ISO 8601 date time format.
+         * @see https://en.wikipedia.org/wiki/ISO_8601
+         */
+        fun formatDateTime(datetime: OffsetDateTime) = DateTimeFormatter.ISO_INSTANT.format(datetime)
+        fun formatDate(date: LocalDate) = DateTimeFormatter.ISO_LOCAL_DATE.format(date)
```

Comment:
@alisters here instead of using the DateFormatters directly, we should use the serializer to keep everything coherent

https://github.com/OpenAPITools/openapi-generator/blob/c27ddc67fe309298b2914c946ceb60b576c09a2b/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt#L254-L262

https://github.com/OpenAPITools/openapi-generator/blob/c27ddc67fe309298b2914c946ceb60b576c09a2b/modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiClient.kt.mustache#L426-L450

Reply:
@4brunu - i have to think through this one. Would this still allow the application of request factory specific adapters if they are added to a single factory instance ?  i agree allowing the multiple serializers is a good move if it can happen.  

I'm on leave for 3 weeks from this afternoon - this work is part work time, part personal time - so i'm not sure when i'll get this done. Are you ok with this potentially restarting in the new year ? I imagine you don't mind if this goes in this release or the next point release ?

Reply:
With the Serializer you can do `Serializer.gsonBuilder.add(...)`

What do you mean by test harness setup?

Sure, when you have time 👍 
Let's merge this when it's more solid, because now we can change whatever we want, once it's merge, we need to worry about backwards compatibility

Reply:
@alisters since this PR was already merge, I would appreciate it if you could continue the changes we discuss here in another PR when you have time 🙂

Reply:
@4brunu - by test harness i just meant someway to run the generated code with the alternate serializers - thats all. I'll raise a new PR for this serializer related work.